File: android/src/main/kotlin/com/klarna/inapp/sdk/flutter_klarna_inapp_sdk/core/stream/BaseStreamHandler.kt
package com.klarna.inapp.sdk.flutter_klarna_inapp_sdk.core.stream

import io.flutter.plugin.common.EventChannel

internal open class BaseStreamHandler<T> : EventChannel.StreamHandler {

    private var sink: EventChannel.EventSink? = null

    override fun onListen(args: Any?, eventSink: EventChannel.EventSink?) {
        sink = eventSink
    }

    override fun onCancel(args: Any?) {
        sink = null
    }

    fun sendValue(value: T) {
        sink?.success(value)
    }

}

File: ios/Classes/Core/Stream/BaseStreamHandler.swift
import Flutter

protocol StreamHandler {
    func sendValue(value: String?)
}

class BaseStreamHandler: NSObject, FlutterStreamHandler, StreamHandler {
    
    private var sink: FlutterEventSink?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
    
    func sendValue(value: String?) {
        sink?(value)
    }
    
    
}
